update.sh

```diff
@@ -33,23 +33,24 @@ hour=$(echo $hour | sed 's/^0*//')
 # Different desktop environment implementations
 case $XDG_CURRENT_DESKTOP in
 	Mint|Mate) setcmd="gsettings set org.mate.background picture-uri";;
-	Cinnamon) setcmd="gsettings set org.cinnamon.background picture-uri";;
-	*) setcmd="gsettings set org.gnome.desktop.background picture-uri";; # GNOME/Unity, default
+	MATE) setcmd="gsettings set org.mate.background picture-filename ";;
+	Cinnamon) setcmd="gsettings set org.cinnamon.background picture-uri file://";;
+	*) setcmd="gsettings set org.gnome.desktop.background picture-uri file://";; # GNOME/Unity, default
 esac
 if [[ -z $XDG_CURRENT_DESKTOP ]]; then # Fallback for i3
 	case $DESKTOP_SESSION in
 		i3) setcmd="feh --bg-fill"
 	esac
 fi
 
-for i in $timing; do # Loop backwards through the wallpapers
-    if (( $hour >= $i )); then
-        $setcmd file://$DIR/${files[i]}
-        echo "Wallpaper set to ${files[i]}"
+for i in ${!timing[@]}; do # Loop through the wallpapers
+    if [ ${timing[$i]} -gt $hour ]; then
+        $setcmd$DIR/${files[i-1]}
+        echo "Wallpaper set to ${files[i-1]}"
```

Comment:
For some reason it wasn't working for me, it would stop at 0 always and then skip to fallback. I'm not very familiar with BASH scripting, so I couldn't figure out what was the issue, so I rewrote it in a way I knew would work.

But please don't take it as an NIH syndrome, I just made the changes so it'd work for me. I think it might just be some distro-specific shenanigans. Let me know what distro you are using and I can start testing against it before my next PR.